### PR TITLE
[MRG] Fix #1424: control smoothing in region extraction

### LINKED ISSUE #1424: smoothing_fwhm passed to connected_regions in RegionExtractor hard coded: problem with mouse data
`smoothing_fwhm` is hard-coded when calling function `connected_regions` in `RegionExtractor`,
https://github.com/nilearn/nilearn/blob/master/nilearn/regions/region_extractor.py#L401
so the default value `smoothing_fwhm=6.` of function `connected_regions` is systematically applied and this is not suitable for my mice images (voxel size is 0.2x0.2x0.2)

Another point is that `min_region_size` is assumed to be an `int` and in my case it may be lower than 1.

I can do a PR to try fix this

### PR DESCRIPTION
**Pull request by salma1601**
Added `smoothing_fwhm` attribute to `RegionExtractor`

### CODE REVIEW

File: nilearn/regions/tests/test_region_extractor.py [hunk 1]
```diff
@@ -155,6 +155,12 @@ def test_region_extractor_fit_and_transform():
     assert_not_equal(extract_ratio.regions_img_, '')
     assert_true(extract_ratio.regions_img_.shape[-1] >= 9)
 
+    # No region for too high FWHM
```
Comment (KamalakerDadi) [inline nilearn/regions/tests/test_region_extractor.py]: I am wandering whether we should test with
 -   ```float``` as input to smoothing_fwhm
- low min_region_size_in_mm3 taking data resolution example same as yours.

Does this sounds good ?

Comment (KamalakerDadi) [inline nilearn/regions/tests/test_region_extractor.py]: >No region for too high FWHM

I would take this as a NOTE in documentation near ```smoothing_fwhm``` and rather not add a test testing whether it raises an error.

I was trying to solve this "zero-size array" problem in PR #1247 by returning am empty array whenever there are no regions found to be extracted.

So, I am not sure whether we should add a test ?

Comment (salma1601) [inline nilearn/regions/tests/test_region_extractor.py]: I changed the test with maps similar to mine, is this ok ?

Comment (salma1601) [inline nilearn/regions/tests/test_region_extractor.py]: Ok, I added this in doc, let me know if it is OK for you

**Comment by KamalakerDadi** [comment]
Thanks. LGTM.

**Comment by KamalakerDadi** [comment]
Track issue at #1424

**Comment by KamalakerDadi** [comment]
Any reviews on this one ?

**Comment by bthirion** [comment]
LGTM.

File: nilearn/regions/tests/test_region_extractor.py [hunk 2]
```diff
@@ -173,6 +173,18 @@ def test_region_extractor_fit_and_transform():
         signal = extractor.transform(img)
         assert_equal(expected_signal_shape, signal.shape)
 
+    # smoke test with high resolution image
+    maps, mask_img = generate_maps((40, 40, 40), n_regions=n_regions,
```
Comment (KamalakerDadi) [inline nilearn/regions/tests/test_region_extractor.py]: I am a bit worried that this image is big, and the test is going to take time. Is there a reason that we don't do it with an image 20x20x20?

Comment (salma1601) [inline nilearn/regions/tests/test_region_extractor.py]: Ok, done

**Comment by GaelVaroquaux** [comment]
LGTM. Thanks. Merging.

**Pull request merged**